### PR TITLE
Coin system skeleton

### LINKED ISSUE #200: Coin System Backend

### PR DESCRIPTION
**Pull request by andrepeiry**
# What ?

A basis for a future coin and currency system in our app !

## How

Add a Wallet property to the user, as well as a dedicated view model for ease of use.

## Closing issues

Close #200 

## Code Coverage

Code Coverage as of 02ba85c6

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  /**
+   * Return the amount of coins currently in the mallet. WARNING: This method does not modify the
+   * value of the coins in the wallet.
+   */
+  fun getCoins(): Coin {
+    return coins
+  }
+
+  /** Creates an empty [Wallet]. It contains no coins. */
+  companion object {
```
Comment (matttsss) [inline app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt:54]: I'm not sure if the `companion object` is realy necesary, we already have a default argument for that?

**Review by matttsss — COMMENTED** [review]
A pretty good start for our community encouragment system. I just have a few doubts about why we would need a anonymous coin system? As you said it is not persistent and the anonymous user can't contribute.

**Comment by matttsss** [comment]
Also, most of the exceptions aren't documented... Could you maybe do something for that?

File: app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt (hunk 1 above)
Comment (andrepeiry) [reply app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt:54]: It is maybe a bit overkill, but as the feature isn't well define in the project, I preferred doing a bit too much than too little (image if, by default, users have a welcome gift, then `Wallet(coin = 0)` make less sens.

Comment (matttsss) [reply app/src/main/java/com/github/se/cyrcle/model/user/Wallet.kt:54]: Right... that makes sense!

**Review by AlexandreBeeDeeBop — CHANGES_REQUESTED** [review]
Are you sure the constant for the threshold should be in the CoinViewModel ? I would think that a better place would be at the instantiation, so that we can better modularize the code.

**Comment by andrepeiry** [comment]
> Are you sure the constant for the threshold should be in the CoinViewModel ? I would think that a better place would be at the instantiation, so that we can better modularize the code.

That's a good thought, I'll update the class to be more modular ^^

**Review by AlexandreBeeDeeBop — APPROVED** [review]
This is getting exciting ! Soon we'll be able to gamble ! 
Jokes aside, nothing else to say, it's all good.

**Review by matttsss — APPROVED** [review]
All good for me, you are free to roll 👌

**Pull request merged**